packages/network-enablement-controller/src/NetworkEnablementController.ts

```diff
@@ -0,0 +1,324 @@
+      s.enabledNetworkMap[namespace][storageKey] = true;
+    });
+  }
+
+  /**
+   * Disables a network for the user.
+   *
+   * This method accepts either a Hex chain ID (for EVM networks) or a CAIP-2 chain ID
+   * (for any blockchain network). The method will automatically convert Hex chain IDs
+   * to CAIP-2 format internally.
+   *
+   * Note: This method will prevent disabling the last remaining enabled network
+   * to ensure at least one network is always available.
+   *
+   * @param chainId - The chain ID of the network to disable. Can be either:
+   * - A Hex string (e.g., '0x1' for Ethereum mainnet) for EVM networks
+   * - A CAIP-2 chain ID (e.g., 'eip155:1' for Ethereum mainnet, 'solana:mainnet' for Solana)
+   */
+  disableNetwork(chainId: Hex | CaipChainId): void {
+    const { namespace, storageKey } = deriveKeys(chainId);
+
+    if (isOnlyNetworkEnabledInNamespace(this.state, namespace, chainId)) {
+      throw new Error('Cannot disable the last remaining enabled network');
+    }
+
+    this.update((s) => {
+      s.enabledNetworkMap[namespace][storageKey] = false;
+    });
+  }
+
+  /**
+   * Ensures that a namespace bucket exists in the state.
+   *
+   * This method creates the namespace entry in the enabledNetworkMap if it doesn't
+   * already exist. This is used to prepare the state structure before adding
+   * network entries.
+   *
+   * @param state - The current controller state
+   * @param ns - The CAIP namespace to ensure exists
+   */
+  #ensureNamespaceBucket(
+    state: NetworkEnablementControllerState,
+    ns: CaipNamespace,
+  ) {
+    if (!state.enabledNetworkMap[ns]) {
+      state.enabledNetworkMap[ns] = {};
+    }
+  }
+
+  /**
+   * Removes a network entry from the state.
+   *
+   * This method is called when a network is removed from the system. It cleans up
+   * the network entry and ensures that at least one network remains enabled.
+   *
+   * @param chainId - The chain ID to remove (Hex or CAIP-2 format)
+   */
+  #removeNetworkEntry(chainId: Hex | CaipChainId): void {
+    const { namespace, storageKey } = deriveKeys(chainId);
+
+    this.update((s) => {
+      // fallback and enable ethereum mainnet
+      if (isOnlyNetworkEnabledInNamespace(this.state, namespace, chainId)) {
+        s.enabledNetworkMap[namespace][ChainId[BuiltInNetworkName.Mainnet]] =
+          true;
+      }
+
+      if (namespace in s.enabledNetworkMap) {
+        delete s.enabledNetworkMap[namespace][storageKey];
+      }
+    });
+  }
+
+  /**
+   * Checks if a network is considered a popular network.
+   *
+   * Popular networks are predefined networks that are commonly used and trusted.
+   * When enabling a non-popular network, the system switches to exclusive mode
+   * (only one network enabled at a time).
+   *
+   * @param caipId - The chain ID to check (can be Hex or CAIP-2 format)
+   * @returns True if the network is popular, false otherwise
+   */
+  #checkIfPopularNetwork(caipId: CaipChainId): boolean {
```

Comment:
What do you think about calling this `#isPopularNetwork` to save on a few characters? 

Actually, does this need to be an method of this class, since it depends on the constant outside the class?